aiogram/contrib/fsm_storage/sqlite.py

```diff
@@ -8,83 +8,98 @@ class SqliteStorage(BaseStorage):
     def __init__(self, db_name: str = 'aiogram_fsm_storage.db', tbl_name: str = 'aiogram_fsm'):
         self._db_name = db_name
         self._tbl_name = tbl_name
-        self._conn = sqlite3.connect(db_name)
-        self._cur = self._conn.cursor()
-
-        self._cur.execute(f'CREATE TABLE IF NOT EXISTS {tbl_name}(chat INTEGER, user INTEGER, state VARCHAR(255), data TEXT, bucket TEXT)')
-        self._conn.commit()
+        self._conn = None
+        
+    async def _get_connect(self) -> aiosqlite.Connection:
+        if self._conn:
+            return self._conn
+        else:
+            self._conn = await aiosqlite.connect(self._db_name)
+            await self._conn.execute(f'CREATE TABLE IF NOT EXISTS {self._tbl_name}(chat INTEGER, user INTEGER, state VARCHAR(255), data TEXT, bucket TEXT)')
+            await self._conn.commit()
+            return self._conn
 
-    def _has_in_db(self, *, chat: typing.Union[str, int, None] = None, user: typing.Union[str, int, None] = None) -> bool:
-        self._cur.execute(f'SELECT EXISTS(SELECT * FROM {self._tbl_name} WHERE chat=? AND user=?)', (chat, user))
-        return self._cur.fetchone()[0]
+    async def _has_in_db(self, *, chat: typing.Union[str, int, None] = None, user: typing.Union[str, int, None] = None) -> bool:
+        conn = await self._get_connect()
+        cursor = await conn.execute(f'SELECT EXISTS(SELECT * FROM {self._tbl_name} WHERE chat=? AND user=?)', (chat, user))
+        return (await cursor.fetchone())[0]
 
     async def close(self):
-        self._conn.close()
+        if self._conn:
+            await self._conn.close()
 
     async def wait_closed(self):
         pass
 
     async def get_state(self, *, chat: typing.Union[str, int, None] = None, user: typing.Union[str, int, None] = None, default: typing.Optional[str] = None) -> typing.Optional[str]:
         chat, user = self.check_address(chat=chat, user=user)
-        if self._has_in_db(chat=chat, user=user):
-            self._cur.execute(f'SELECT state FROM {self._tbl_name} WHERE chat=? AND user=?', (chat, user))
-            return self._cur.fetchone()[0]
+        conn = await self._get_connect()
+        if await self._has_in_db(chat=chat, user=user):
```

Comment:
Isolate `get_connect` levels. You're trying to get it again and again, but you already have one...

Reply:
I isolated `get_connect` as `conn = self._conn or await self._get_connect()`